pkg/operator/ceph/csi/spec.go

```diff
@@ -676,12 +677,39 @@ func (r *ReconcileCSI) startDrivers(ver *version.Info, ownerInfo *k8sutil.OwnerI
 	return nil
 }
 
-func (r *ReconcileCSI) stopDrivers(ver *version.Info) error {
+// return false is rook operator is not the owner of csi driver,
+// so csi driver do not get deleted as it isn't deployed by
+// rook operator and someone externally installed it
+func (r *ReconcileCSI) driverOwnedByRook(ownerInfo *k8sutil.OwnerInfo, deploymentName string) bool {
+	getDeployment, err := r.context.Clientset.AppsV1().Deployments(r.opConfig.OperatorNamespace).Get(r.opManagerContext, deploymentName, metav1.GetOptions{})
+	if err != nil {
+		logger.Warningf("csi deployment pod not present %q %q", deploymentName, r.opConfig.OperatorNamespace)
+		return true
+	}
+
+	var operatorOwnerInfo, csiDriverOwnerInfo *metav1.OwnerReference
+	if ownerInfo != nil {
+		operatorOwnerInfo = ownerInfo.GetOwnerReference()
+	}
+	if getDeployment.OwnerReferences[0] != (metav1.OwnerReference{}) {
+		csiDriverOwnerInfo = &getDeployment.OwnerReferences[0]
+	}
+
+	// check the value not the address
+	if reflect.DeepEqual(csiDriverOwnerInfo, operatorOwnerInfo) {
+		logger.Debugf("csi owner reference %v not matches operator owner reference  %v, so don't delete the csi driver", operatorOwnerInfo, csiDriverOwnerInfo)
```

Comment:
I am getting this, both values seems same, not sure what is missing @Madhu-1 ?
```
ceph-csi: csi owner reference &OwnerReference{Kind:Deployment,Name:rook-ceph-operator,UID:f3d4cba9-3392-4d51-a7e4-f69c697ccbaf,APIVersion:apps/v1,Controller:*true,BlockOwnerDeletion:*false,} not matches operator owner reference  &OwnerReference{Kind:Deployment,Name:rook-ceph-operator,UID:f3d4cba9-3392-4d51-a7e4-f69c697ccbaf,APIVersion:apps/v1,Controller:*true,BlockOwnerDeletion:*false,}, so don't delete the csi driver

```

Reply:
I think you should negate the condition.

Reply:
it worked with my latest changes,
@leelavg would you like to test the image in your external deployment case `quay.io/paarora/rook-operator:v492`

Reply:
or I will ping with the latest image when all the changes are done